sys/kern/src/arch/rv64/task.rs

```diff
@@ -54,19 +73,30 @@ pub fn start_first_task(tick_divisor: u32, task: &mut task::Task) -> ! {
         set_timer(tick_divisor - 1);
 
         // Machine timer interrupt enable
-        register::mie::set_mtimer();
+        set_xtimer();
     }
 
     // Load first task pointer, set its initial stack pointer, and exit out
     // of machine mode, launching the task.
     unsafe {
         crate::task::activate_next_task(task);
-        asm!("
-            ld sp, ({0})
-            mret",
-            in(reg) &task.save().sp(),
-            options(noreturn)
-        );
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "riscv-supervisor-mode")] {
```

Comment:
Can we just string format the "s" vs "m", rather than cfg_if the whole block?

Reply:
Fixed this one with concat!(...)